src/main/java/com/assignment/StudentInfo.java

```diff
@@ -0,0 +1,103 @@
+package com.assignment;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+
+public class StudentInfo {
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+    ArrayList<Student> studentList = new ArrayList<>();
+
+    public void setStudentInfo(String[] info) {
+        Student student = new Student();
+
+        student.setId(info[0]);
+        student.setName(info[1]);
+        student.setAddress(info[2]);
+        student.setStudentClass(info[3]);
+        student.setSchool(info[4]);
+        studentList.add(student);
+        logger.info("Student information added");
+    }
+
+    public int getStudentList() {
+        Student student = new Student();
+        int count = 0;
+        if (studentList.size() != 0) {
+            for (int i = 0; i < studentList.size(); i++) {
+                student = studentList.get(i);
+                logger.info(student.getId() + " " + student.getName() +
+                        " " + student.getAddress() + " " + student.getStudentClass() +
+                        " " + student.getSchool());
+                count++;
+            }
+        } else {
+            logger.info("There is no student information");
+        }
+        return count;
+    }
+
+    public Student getStudentInfo(String studentDetails) {
+        Student student = new Student();
+        int count = 0;
+        for (int i = 0; i < studentList.size(); i++) {
+            student = studentList.get(i);
+            if (student.getId() == studentDetails || student.getName() == studentDetails
+                    || student.getAddress() == studentDetails || student.getStudentClass() == studentDetails
+                    || student.getSchool() == studentDetails) {
```

Comment:
In studentDetails variable you are passing only **id**, so in if condition why are you checking student.getName() == studentDetails  || student.getAddress() == studentDetails || student.getStudentClass() == studentDetails
|| student.getSchool() == studentDetails ???